panoptes_client/panoptes.py

```diff
@@ -231,9 +233,16 @@ def get_csrf_token(self):
 
     def get_bearer_token(self):
         if not self.bearer_token or self.bearer_expires > datetime.now():
-            if not self.logged_in:
+
+            if self.client_secret:
+                grant_type = 'client_credentials'
+            elif self.password:
+                grant_type = 'password'
+
+            if not (self.logged_in and grant_type == 'client_credentials'):
                 if not self.login():
                     return
```

Comment:
Presumably there's no username/password when you're using client_credentials, in which case this will always return here because `self.login()` will return `None` at line 201.